modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2077,8 +2077,7 @@ def sort_function(df):
             new_partitions,
             new_lengths[axis.value],
         ) = self._partition_mgr_cls.rebalance_partitions(new_partitions)
-        col_parts = new_partitions[0]
-        new_lengths[axis.value ^ 1] = [part.width() for part in col_parts]
+        new_lengths[axis.value ^ 1] = [len(self.columns)]
```

Comment:
we should add a comment explaining the assumption upon which we hardcode this (that we're doing the sort in a full-axis way)